src/prefect/cli/deployment.py

```diff
@@ -501,64 +506,155 @@ async def delete_schedule(
         exit_with_success(f"Deleted deployment schedule {schedule_id}")
 
 
-@schedule_app.command("pause")
-async def pause_schedule(deployment_name: str, schedule_id: UUID):
-    """
-    Pause a deployment schedule.
-    """
-    assert_deployment_name_format(deployment_name)
+async def _update_schedule_status(
+    deployment_name: Optional[str],
+    schedule_id: Optional[UUID],
+    _all: bool,
+    active: bool,
+    action: str,  # "pause" or "resume"
+) -> None:
+    """Shared logic for pausing and resuming deployment schedules."""
+    past_tense = "paused" if not active else "resumed"
+    present_tense = "pause" if not active else "resume"
 
-    async with get_client() as client:
-        try:
-            deployment = await client.read_deployment_by_name(deployment_name)
-        except ObjectNotFound:
-            return exit_with_error(f"Deployment {deployment_name!r} not found!")
+    if _all:
+        if deployment_name is not None or schedule_id is not None:
+            return exit_with_error(
+                "Cannot specify deployment name or schedule ID with --all"
+            )
```

Comment:
This feels like it shouldn't be nested and should be an early exit at the top of this function.